amaranth/lib/wiring.py

```diff
@@ -570,6 +570,8 @@ def __getattr__(self, name):
         value = getattr(self.__unflipped, name)
         if inspect.ismethod(value):
             return types.MethodType(value.__func__, self)
+        elif isinstance(value, Interface) and name in self.__unflipped.signature.members:
```

Comment:
I don't think that's right. An interface object isn't an instance of `Interface`, it's an object with a `signature` attribute that is a signature. I think you should just check the type of the member instead.